services/database-analyzer-sequelize.js

```diff
@@ -0,0 +1,171 @@
+            foreignKeyName: _.camelCase(foreignKey.column_name),
+            as: formatAliasName(foreignKey.column_name),
+          };
+
+          // NOTICE: If the foreign key name and alias are the same, Sequelize will crash, we need
+          //         to handle this specific scenario generating a different foreign key name.
+          if (reference.foreignKeyName === reference.as) {
+            reference.foreignKeyName = `${reference.foreignKeyName}Key`;
+          }
+
+          if (foreignKey.foreign_column_name !== 'id') {
+            reference.targetKey = foreignKey.foreign_column_name;
+          }
+
+          references.push(reference);
+        } else if (type) {
+          // NOTICE: If the column is of integer type, named "id" and primary, Sequelize will
+          //         handle it automatically without necessary declaration.
+          if (!(nameColumn === 'id' && type === 'INTEGER' && columnInfo.primaryKey)) {
+            const field = {
+              name: _.camelCase(nameColumn),
+              nameColumn,
+              type,
+              primaryKey: columnInfo.primaryKey,
+              defaultValue: columnInfo.defaultValue,
+            };
+
+            fields.push(field);
+          }
+        }
+      });
+
+      const options = {
+        underscored: isUnderscored(fields),
+        timestamps: hasTimestamps(fields),
+        hasIdColumn: hasIdColumn(fields, primaryKeys),
+        hasPrimaryKeys: !_.isEmpty(primaryKeys),
+      };
+
+      return {
+        fields,
+        references,
+        primaryKeys,
+        options,
+      };
+    });
+}
+
+async function sequelizeTableAnalyzer(databaseConnection, config, allowWarning) {
+  const schema = {};
+
+  queryInterface = databaseConnection.getQueryInterface();
+  tableForeignKeysAnalyzer = new TableForeignKeysAnalyzer(databaseConnection, config);
+  columnTypeGetter = new ColumnTypeGetter(databaseConnection, config.dbSchema || 'public', allowWarning);
+
+  if (config.dbSchema) {
+    const schemaExists = await queryInterface.sequelize
+      .query(
+        'SELECT SCHEMA_NAME FROM INFORMATION_SCHEMA.SCHEMATA WHERE SCHEMA_NAME = ?;',
+        { type: queryInterface.sequelize.QueryTypes.SELECT, replacements: [config.dbSchema] },
+      )
+      .then(result => !!result.length);
+
+    if (!schemaExists) {
+      logger.error('This schema does not exists.');
+      return process.exit(1);
+    }
+  }
+
+  // Build the db schema.
+  await P.mapSeries(queryInterface.showAllTables({
+    schema: config.dbSchema,
+  }), async (table) => {
+    // NOTICE: MS SQL returns objects instead of strings.
+    if (typeof table === 'object') {
+    // eslint-disable-next-line no-param-reassign
+      table = table.tableName;
+    }
+
+    schema[table] = await analyzeTable(table, config);
+  });
+
+  if (_.isEmpty(schema)) {
+    throw new DatabaseAnalyzerError.EmptyDatabase('sequelize', databaseConnection.getDialect());
```

Comment:
[Regression?] I don't know if you tested it or not but it looks broken. The catch won't understand the `details.orm` and `details.dialect` from what I can see.

And to confirm that, the mongoose case seems to have the right arguments according to your new implementation.
```javascript
new DatabaseAnalyzerError.EmptyDatabase('no collections found', {
  orm: 'mongoose',
  dialect: 'mongodb',
});
```

Reply:
Oh yes, the parameters are not OK. 
I tested it, and it displays:
`> Your database looks empty! Please create some tables before running the command.`

That was OK for me but you are right, it contains a regression.
Thanks